core/src/domain/interactor/DeleteAllInteractor.php

```diff
@@ -25,17 +20,15 @@ public function __construct(DeleteRepository $deleteRepository)
     }
 
     /**
-     * Execute
-     *
-     * @param Query     $query     query
-     * @param Operation $operation operation
+     * @param Query     $query
+     * @param Operation $operation
      *
-     * @return void
+     * @return bool
      */
     public function execute(
         Query $query,
         Operation $operation
-    ) {
-        $this->deleteRepository->deleteAll($query, $operation);
+    ): bool {
+        return $this->deleteRepository->deleteAll($query, $operation);
     }
```

Comment:
What's the purpose of returning a bool here?

Reply:
It's just a detail but I think it would be better to just return void in case of success and a domain exception in case an error so you can then handle different types of errors instead of returning true/false which is very limiting in terms on error handling.

Reply:
Everything else looks pretty good, I am looking forward to implementing it in SP. I'll probably have more feedback after I start using it.

Reply:
Agree, changed bool to void return.